### ISSUE TITLE
React multiple select dropdown with checkboxes, when open, has critical and serious accessibility errors.

### ISSUE THREAD

**Issue by estrizhiver** [mui/mui-x#13197]
### Search keywords

Axe devtools, accessibility errors

### Latest version

- [X] I have tested the latest version

### Steps to reproduce

Link to live example: https://mui.com/material-ui/react-select/#checkmarks

Steps:
1. Open dropdown
2. Run Axe devtools with dropdown open

### Current behavior

Critical and serious errors should not exist. We are seeing the same errors in our production environment. I'll provide screenshots for both mui and our environments.

MUI

Critical errors:
<img width="1510" alt="image" src="https://github.com/mui/mui-x/assets/7647075/1ef73394-ddf9-4a9d-be37-353b7187ed7c">

Serious errors:
<img width="1510" alt="image" src="https://github.com/mui/mui-x/assets/7647075/1885da1f-705f-414f-b2bc-6b703576f915">

ProcessUnity

Critical:
<img width="1510" alt="image" src="https://github.com/mui/mui-x/assets/7647075/c535ff80-d74d-4105-91aa-94bf7455762d">

Serious:
<img width="1510" alt="image" src="https://github.com/mui/mui-x/assets/7647075/85bd33ff-4f01-4b5f-b852-406593324a82">

### Expected behavior

There should be no accessibility errors when dropdown is open.

### Context

Eliminate accessibility errors and become a11y compliant.

### Your environment

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Chrome on a Mac
  Output from `npx @mui/envinfo` goes here.
```
</details>

### Order ID or Support key 💳 (optional)

8005
